csrc/scheduler/hopper_multi_matmul.cpp

```diff
@@ -305,11 +305,11 @@ MmaInputSmemSwizzle tmaSwizzleSharedMemory(TensorView* shared_mem_tv) {
   const int64_t B64_elements = 64 / dataTypeSize(dtype);
   const int64_t B32_elements = 32 / dataTypeSize(dtype);
 
-  if (inner_dim_size >= B128_elements) {
+  if (inner_dim_size % B128_elements == 0) {
     return MmaInputSmemSwizzle::B128;
-  } else if (inner_dim_size >= B64_elements) {
+  } else if (inner_dim_size % B64_elements == 0) {
     return MmaInputSmemSwizzle::B64;
-  } else if (inner_dim_size >= B32_elements) {
+  } else if (inner_dim_size % B32_elements == 0) {
```

Comment:
I think we should add tests with odd CTA tile N like 24 and 48, because with this change I believe we will hit the assert.